manager/state/raft/transport/peer.go

```diff
@@ -142,7 +198,35 @@ func (p *peer) sendProcessMessage(ctx context.Context, m raftpb.Message) error {
 	}
 	ctx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
-	_, err := api.NewRaftClient(p.conn()).ProcessRaftMessage(ctx, &api.ProcessRaftMessageRequest{Message: &m})
+
+	var err error
+	if needsStreaming(&m) {
+		// stream message if its bigger than MaxRaftMsgSize.
+		var stream api.Raft_RaftMessageStreamClient
+		stream, err = api.NewRaftClient(p.conn()).RaftMessageStream(ctx)
+		if err == nil {
+			// split message
+			msgs := splitSnapshotData(ctx, &m)
+			for _, msg := range msgs {
+				err = stream.Send(&msg)
+				if err != nil {
+					log.G(ctx).WithError(err).Error("failed to stream message")
+				}
+			}
+			_, err = stream.CloseAndRecv()
+
+			// TODO(anshul) Check if Unimplemented and bubble up the error.
+			if err != nil {
+				log.G(ctx).WithError(err).Error("error receiving response on stream close")
+				return err
+			}
+		} else {
+			log.G(ctx).WithError(err).Error("failed to create stream!")
```

Comment:
Idea:

```go
// if the RaftMessageStream method is not implemented on the server, log that and try again
if status.FromError(err).Code() == codes.Unimplemented {
    log.G(ctx).WithError(err).Error("peer does not support streaming (probably an older version)")
    _, err = api.NewRaftClient(p.conn()).ProcessRaftMessage(ctx, &api.ProcessRaftMessageRequest{Message: &m})
} else {
    log.G(ctx).WithError(err).Error("failed to create stream")
}
```

or something similar to that. you'll need to double-check that this works the way I think it does. Also, when you re-implement to always prefer streaming, you'll need this check anyway, because you'll try streaming first and then try not streaming after.